src/WebJobs.Script/RoutingUtility.cs

```diff
@@ -0,0 +1,198 @@
+            {typeof(GuidRouteConstraint), (str => Guid.Parse(str)) },
+            {typeof(IntRouteConstraint), (str => Convert.ToInt32(str)) },
+            {typeof(LongRouteConstraint), (str => Convert.ToInt64(str)) },
+            {typeof(RangeRouteConstraint), (str => Convert.ToInt64(str)) }
+        };  
+
+        private static TemplateMatcher GetTemplateMatcher(string queryTemplate)
+        {
+            TemplateMatcher templateMatcher = null;
+            Templates.TryGetValue(queryTemplate, out templateMatcher);
+            if (templateMatcher == null)
+            {
+                var routeTemplate = TemplateParser.Parse(queryTemplate);
+                var defaults = new RouteValueDictionary();
+                foreach (var parameter in routeTemplate.Parameters)
+                {
+                    if (parameter.DefaultValue != null)
+                    {
+                        defaults.Add(parameter.Name, parameter.DefaultValue);
+                    }
+                }
+                SetConstraints(queryTemplate, routeTemplate);
+                templateMatcher = new TemplateMatcher(routeTemplate, defaults);
+                Templates.Add(queryTemplate, templateMatcher);
+
+            }
+            return templateMatcher;
+        }
+
+        private static void SetConstraints(string queryTemplate, RouteTemplate parsedTemplate)
+        {
+            //if the constraints for this route have not already been set, add them to ConstraintsMap
+            IDictionary<string, IRouteConstraint> constraints = null;
+            ConstraintsMap.TryGetValue(queryTemplate, out constraints);
+            if (constraints == null)
+            {
+                var constraintBuilder = new RouteConstraintBuilder(ConstraintResolver, parsedTemplate.TemplateText);
+                foreach (var parameter in parsedTemplate.Parameters)
+                {
+                    if (parameter.IsOptional)
+                    {
+                        constraintBuilder.SetOptional(parameter.Name);
+                    }
+
+                    foreach (var inlineConstraint in parameter.InlineConstraints)
+                    {
+                        constraintBuilder.AddResolvedConstraint(parameter.Name, inlineConstraint.Constraint);
+                    }
+                }
+                constraints = constraintBuilder.Build();
+                ConstraintsMap.Add(queryTemplate, constraints);
+            }
+        }
+
+
+        public static void ClearTemplates()
+        {
+            Templates.Clear();
+            ConstraintsMap.Clear();
+        }
+
+        public static bool MatchesTemplate(string queryTemplate, string query)
+        {
+            try
+            {
+                var templateMatcher = GetTemplateMatcher(queryTemplate);
+                var pathString = new PathString("/" + query);
+                var values = new RouteValueDictionary();
```

Comment:
Per our conversation offline, add a comment to let the reader know how 'values' is being populated.